### PR TITLE
docs(lib-builder): Add docker image documentation

### PR DESCRIPTION
**Pull request by lucasssvaz**
## Description of Change
Add documentation for the Lib Builder Docker Image.

### CODE REVIEW

File: docs/en/lib_builder.rst [hunk 1]
```diff
@@ -152,8 +152,10 @@ This build command will build for the ESP32-S3 target. You can specify other tar
 
 * esp32
 * esp32s2
-* esp32c3
 * esp32s3
+* esp32c2
+* esp32c3
+* esp32h2
```
Comment (me-no-dev) [inline docs/en/lib_builder.rst:158]: esp32c6?

**Comment by lucasssvaz** [comment]
@P-R-O-C-H-Y PTAL at why Wokwi tests are failing.

**Review by me-no-dev — APPROVED** [review]

File: docs/en/lib_builder.rst [hunk 2]
```diff
@@ -169,11 +172,130 @@ Set the build type. ex. 'build' to build the project and prepare for uploading t
...
+- ``-e TERM=xterm-256color``: Optional. Sets the terminal type to ``xterm-256color`` to display colors correctly;
+- ``-v $PWD:/arduino-esp32``: Optional. Mounts the current folder at ``/arduino-esp32`` inside the container. If not provided, the container will not copy the compiled libraries to the host machine;
+- ``espressif/esp32-arduino-lib-builder``: uses Docker image ``espressif/esp32-arduino-lib-builder`` with tag ``latest``. The ``latest`` tag is implicitly added by Docker when no tag is specified.
+
+.. warning::
+    The ``-v`` option is used to mount a folder from the host machine to the container. Make sure the folder already exists on the host machine before running the command.
+    Otherwise, the folder will be created with root permissions and files generated inside the container might cause permission issues and compilation errors.
+
+.. note::
+   When the mounted directory, ``/arduino-esp32``, contains a git repository owned by a different user (``UID``) than the one running the Docker container,
```
Comment (P-R-O-C-H-Y) [inline docs/en/lib_builder.rst:245]:
```suggestion
   When the mounted directory ``/arduino-esp32`` contains a git repository owned by a different user (``UID``) than the one running the Docker container,
```

File: docs/en/lib_builder.rst [hunk 3]
```diff
@@ -169,11 +172,130 @@ Set the build type. ex. 'build' to build the project and prepare for uploading t
...
+   git commands executed within ``/arduino-esp32`` might fail, displaying an error message ``fatal: detected dubious ownership in repository at '/arduino-esp32'``.
+   To resolve this issue, you can designate the ``/arduino-esp32`` directory as safe by setting the ``LIBBUILDER_GIT_SAFE_DIR`` environment variable during the Docker container startup.
+   For instance, you can achieve this by including ``-e LIBBUILDER_GIT_SAFE_DIR='/arduino-esp32'`` as a parameter. Additionally, multiple directories can be specified by using a ``:`` separator.
+   To entirely disable this git security check, ``*`` can be used.
+
+After running the above command, you will be inside the container and can build the libraries using the user interface.
```
Comment (P-R-O-C-H-Y) [inline docs/en/lib_builder.rst:251]:
```suggestion
After running the above command, you will be inside the container and the libraries can be built using the user interface.
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
Small nitpicks

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**